subsys/bluetooth/controller/ll_sw/lll.h

```diff
@@ -151,96 +152,49 @@ struct lll_event {
-#if defined(CONFIG_BT_CTLR_SYNC_PERIODIC)
-	NODE_RX_TYPE_SYNC = 0x09,
-	NODE_RX_TYPE_SYNC_LOST = 0x0a,
-#endif /* CONFIG_BT_CTLR_SYNC_PERIODIC */
-#endif /* CONFIG_BT_CTLR_ADV_EXT */
-#endif /* CONFIG_BT_OBSERVER */
-
-#if defined(CONFIG_BT_BROADCASTER)
-#if defined(CONFIG_BT_CTLR_ADV_EXT)
-	NODE_RX_TYPE_EXT_ADV_TERMINATE = 0x0b,
-#endif /* CONFIG_BT_CTLR_ADV_EXT */
-#endif /* CONFIG_BT_BROADCASTER */
-
-#if defined(CONFIG_BT_CTLR_SCAN_REQ_NOTIFY)
-	NODE_RX_TYPE_SCAN_REQ = 0x0c,
-#endif /* CONFIG_BT_CTLR_SCAN_REQ_NOTIFY */
-
-#if defined(CONFIG_BT_CONN)
-	NODE_RX_TYPE_CONNECTION = 0x0d,
-	NODE_RX_TYPE_TERMINATE = 0x0e,
-	NODE_RX_TYPE_CONN_UPDATE = 0x0f,
-	NODE_RX_TYPE_ENC_REFRESH = 0x10,
-
-#if defined(CONFIG_BT_CTLR_LE_PING)
-	NODE_RX_TYPE_APTO = 0x11,
-#endif /* CONFIG_BT_CTLR_LE_PING */
-
-	NODE_RX_TYPE_CHAN_SEL_ALGO = 0x12,
-
-#if defined(CONFIG_BT_CTLR_PHY)
-	NODE_RX_TYPE_PHY_UPDATE = 0x13,
-#endif /* CONFIG_BT_CTLR_PHY */
-
-#if defined(CONFIG_BT_CTLR_CONN_RSSI)
-	NODE_RX_TYPE_RSSI = 0x14,
-#endif /* CONFIG_BT_CTLR_CONN_RSSI */
-#endif /* CONFIG_BT_CONN */
-
-#if defined(CONFIG_BT_CTLR_PROFILE_ISR)
-	NODE_RX_TYPE_PROFILE = 0x15,
-#endif /* CONFIG_BT_CTLR_PROFILE_ISR */
-
-#if defined(CONFIG_BT_CTLR_ADV_INDICATION)
-	NODE_RX_TYPE_ADV_INDICATION = 0x16,
-#endif /* CONFIG_BT_CTLR_ADV_INDICATION */
+	NODE_RX_TYPE_REPORT,
+	NODE_RX_TYPE_EXT_1M_REPORT,
+	NODE_RX_TYPE_EXT_2M_REPORT,
+	NODE_RX_TYPE_EXT_CODED_REPORT,
+	NODE_RX_TYPE_EXT_SCAN_TERMINATE,
+	NODE_RX_TYPE_SYNC,
+	NODE_RX_TYPE_SYNC_LOST,
+	NODE_RX_TYPE_EXT_ADV_TERMINATE,
+	NODE_RX_TYPE_SCAN_REQ,
+	NODE_RX_TYPE_CONNECTION,
+	NODE_RX_TYPE_TERMINATE,
+	NODE_RX_TYPE_CONN_UPDATE,
+	NODE_RX_TYPE_ENC_REFRESH,
+	NODE_RX_TYPE_APTO,
+	NODE_RX_TYPE_CHAN_SEL_ALGO,
+	NODE_RX_TYPE_PHY_UPDATE,
+	NODE_RX_TYPE_RSSI,
+	NODE_RX_TYPE_PROFILE,
+	NODE_RX_TYPE_ADV_INDICATION,
+	NODE_RX_TYPE_SCAN_INDICATION,
+	NODE_RX_TYPE_CIS_REQUEST,
+	NODE_RX_TYPE_CIS_ESTABLISHED,
+	NODE_RX_TYPE_MESH_ADV_CPLT,
+	NODE_RX_TYPE_MESH_REPORT,
 
-#if defined(CONFIG_BT_CTLR_SCAN_INDICATION)
-	NODE_RX_TYPE_SCAN_INDICATION = 0x17,
-#endif /* CONFIG_BT_CTLR_SCAN_INDICATION */
-
-#if defined(CONFIG_BT_CTLR_PERIPHERAL_ISO)
-	NODE_RX_TYPE_CIS_REQUEST = 0x18,
-#endif /* CONFIG_BT_CTLR_PERIPHERAL_ISO */
-
-#if defined(CONFIG_BT_CTLR_CENTRAL_ISO) || \
-	defined(CONFIG_BT_CTLR_PERIPHERAL_ISO)
-	NODE_RX_TYPE_CIS_ESTABLISHED = 0x19,
-#endif /* CONFIG_BT_CTLR_CENTRAL_ISO || CONFIG_BT_CTLR_PERIPHERAL_ISO */
-
-#if defined(CONFIG_BT_HCI_MESH_EXT)
-	NODE_RX_TYPE_MESH_ADV_CPLT = 0x1a,
-	NODE_RX_TYPE_MESH_REPORT = 0x1b,
-#endif /* CONFIG_BT_HCI_MESH_EXT */
-
-/* Following proprietary defines must be at end of enum range */
 #if defined(CONFIG_BT_CTLR_USER_EXT)
-	NODE_RX_TYPE_USER_START = 0x1c,
-	NODE_RX_TYPE_USER_END = NODE_RX_TYPE_USER_START +
-				CONFIG_BT_CTLR_USER_EVT_RANGE,
+	/* No entries shall be added after the NODE_RX_TYPE_USER_START/END */
+	NODE_RX_TYPE_USER_START,
+	UTIL_LISTIFY(CONFIG_BT_CTLR_USER_EVT_RANGE, DEFINE_NODE_RX_USER_TYPE, _)
```

Comment:
I think there is an off-by-one error here. If `CONFIG_BT_CTLR_USER_EVT_RANGE`=0, `NODE_RX_TYPE_USER_END` should get the same value as `NODE_RX_TYPE_USER_START` but it will get the value+1 with this change.

Reply:
I don't think that will be a problem, `NODE_RX_TYPE_USER_END` is only used in the case `NODE_RX_TYPE_USER_START ... NODE_RX_TYPE_USER_END`, so it catches an extra value, so you'll get CONFIG_BT_CTLR_USER_EVT_RANGE+1 user defined node rx types.

Reply:
I guess changing the cases to `NODE_RX_TYPE_USER_START ... NODE_RX_TYPE_USER_END-1` would avoid the off-by-one then.

Reply:
@wopu-ot We can also guard this part with `CONFIG_BT_CTLR_USER_EVT_RANGE > 0` and then change the list to 
```
	UTIL_LISTIFY(CONFIG_BT_CTLR_USER_EVT_RANGE, DEFINE_NODE_RX_USER_TYPE, _)
```

Would you prefer that?

Reply:
UTIL_LISTIFY doesn't work with numeric expressions.

Reply:
I see. I guess a hidden Kconfig that is 1 less than `CONFIG_BT_CTLR_USER_EVT_RANGE` would "fix" that, but then we're just walking towards making things unnecessary complicated IMO. 

Or do as suggested with `NODE_RX_TYPE_USER_START ... NODE_RX_TYPE_USER_END - 1`

Reply:
I prefer the latter. Counting/iterating from 0 to N-1 is a perfectly normal and non-hacky way of doing things.

Reply:
Updated the PR. 

Speaking of off-by-one, I wonder if that is still the case (and always was), since if `CONFIG_BT_CTLR_USER_EVT_RANGE=0` we still have 1 case in the switches that would match (if `type == NODE_RX_TYPE_USER_START`). 

Should we guard 
```
case NODE_RX_TYPE_USER_START ... NODE_RX_TYPE_USER_END - 1:
```
with 
```
#if CONFIG_BT_CTLR_USER_EVT_RANGE > 0
```
?

Reply:
That would be a good idea